torch/_inductor/fx_passes/micro_pipeline_tp.py

```diff
@@ -771,58 +844,57 @@ def fuse_matmul_reduce_scatter(reduce_scatter: _ReduceScatterMatch) -> bool:
 
     Returns boolean indicating if fusion was successful or not.
     """
-    if (
-        not torch.distributed.is_available()
-        or not torch.distributed.is_nccl_available()
-    ):
-        log.debug(
-            "torch.distributed is not available, skipping fuse_matmul_reduce_scatter fusion"
-        )
-        return False
+    assert torch.distributed.is_available() or torch.distributed.is_nccl_available(), (
+        "torch.distributed must be available to use async tensor parallelism"
```

Comment:
And should we change to be more precisely to indicate we require "NCCL"?

Reply:
Updated the assertion message to clarify NCCL is required as well